mne_bids/tests/write/test_write.py

```diff
@@ -2543,137 +2536,6 @@ def test_sidecar_encoding(_bids_validate, tmpdir):
-@pytest.mark.filterwarnings(warning_str['channel_unit_changed'])
-def test_convert_brainvision(dir_name, fname, reader, _bids_validate, tmpdir):
-    """Test conversion of EEG/iEEG manufacturer format to BrainVision.
-
-    BrainVision should correctly store data from pybv>=0.5 that
-    has different non-voltage units.
-    """
-    bids_root = tmpdir.mkdir('bids1')
-    data_path = op.join(testing.data_path(), dir_name)
-    raw_fname = op.join(data_path, fname)
-
-    # the BIDS path for test datasets to get written to
-    bids_path = _bids_path.copy().update(root=bids_root, datatype='eeg')
-
-    raw = reader(raw_fname)
-    kwargs = dict(raw=raw, bids_path=bids_path, overwrite=True)
-
-    # alter some channels manually; in NK and Persyst, this will cause
-    # channel to not have units
-    raw.set_channel_types({raw.info['ch_names'][0]: 'stim'})
-
-    if dir_name == 'NihonKohden':
-        with pytest.warns(RuntimeWarning,
-                          match='Encountered data in "short" format'):
-            bids_output_path = write_raw_bids(**kwargs)
-    else:
-        with pytest.warns(RuntimeWarning,
-                          match='Encountered data in "double" format'):
-            bids_output_path = write_raw_bids(**kwargs)
-
-    # channel units should stay the same
-    raw2 = read_raw_bids(bids_output_path)
-    assert all([ch1['unit'] == ch2['unit'] for ch1, ch2 in
-                zip(raw.info['chs'], raw2.info['chs'])])
-    assert raw2.info['chs'][0]['unit'] == FIFF.FIFF_UNIT_NONE
-
-    # load in the channels tsv and the channel unit should be not set
-    channels_fname = bids_output_path.update(
-        suffix='channels', extension='.tsv')
-    channels_tsv = _from_tsv(channels_fname)
-    assert channels_tsv['units'][0] == 'n/a'
-
-
-@requires_version('mne', '0.22')
-@pytest.mark.parametrize(
-    'dir_name, fname, reader', test_convertbrainvision_data)
-@pytest.mark.filterwarnings(warning_str['channel_unit_changed'])
-def test_error_write_meg_as_eeg(dir_name, fname, reader, tmpdir):
-    """Test error writing as BrainVision EEG data for MEG."""
-    bids_root = tmpdir.mkdir('bids1')
-    data_path = op.join(testing.data_path(), dir_name)
-    raw_fname = op.join(data_path, fname)
-
-    bids_path = _bids_path.copy().update(root=bids_root, datatype='eeg',
-                                         extension='.vhdr')
-    raw = reader(raw_fname)
-    kwargs = dict(raw=raw, bids_path=bids_path.update(datatype='meg'))
-
-    # if we accidentally add MEG channels, then an error will occur
-    raw.set_channel_types({raw.info['ch_names'][0]: 'mag'})
-    with pytest.raises(ValueError, match='Got file extension .*'
-                                         'for MEG data'):
-        write_raw_bids(**kwargs)
-
-
-@pytest.mark.parametrize('dir_name, fname, reader', test_convert_data)
-@pytest.mark.filterwarnings(warning_str['channel_unit_changed'])
-def test_convert_dataset_format(dir_name, fname, reader, tmpdir):
```

Comment:
These error checks and assertions could be moved into the existing tests actually.